script.js

```diff
@@ -1,12 +1,210 @@
+  const selectMenu = document.getElementById(_epDDLCtrlId);
+  selectMenu.innerHTML = "";
+
+  const optionPlaceHolderEl = document.createElement("option");
+  optionPlaceHolderEl.innerText = "--Please Select--";
+  selectMenu.appendChild(optionPlaceHolderEl);
+
+  _localEpisodes.forEach((episode) => {
+    const optionEl = document.createElement("option");
+    optionEl.innerText = `S${formatVal(episode.season)}E${formatVal(
+      episode.number
+    )} - ${episode.name}`;
+
+    selectMenu.appendChild(optionEl);
+  });
+  drawEpiCard();
+}
+function refreshShowMenu() {
+  const selectShow = document.getElementById(_showDDLCtrlId);
+  selectShow.innerHTML = "";
+  //add options to ddl(select show)
+  _localShows.forEach((show) => {
+    const showEl = document.createElement("option");
+    showEl.innerText = show.name;
+    showEl.value = show.id;
+    selectShow.appendChild(showEl);
+  });
+}
+function createSearchResult() {
+  const rootElem = document.getElementById("root");
+  const resultP = document.createElement("p");
+  resultP.innerHTML = `Got ${_localEpisodes.length} episode(s). The data has (originally) come from <a href= "https://www.tvmaze.com/">TVMaze.com</a><br>`;
+  rootElem.appendChild(resultP);
+}
+function createShowMenu() {
+  const rootElem = document.getElementById("root");
+
+  // select/ddl (select show)
+  const selectShow = document.createElement("select");
+  selectShow.id = _showDDLCtrlId;
+  selectShow.addEventListener("change", function () {
+    refreshEpisodesDDL(selectShow.value);
+  });
+
+  rootElem.appendChild(selectShow);
 }
+function createEPMenu() {
+  const rootElem = document.getElementById("root");
+  const selectMenu = document.createElement("select");
+  selectMenu.id = _epDDLCtrlId;
+  selectMenu.addEventListener("change", function () {
+    drawEpiCard();
+  });
+  rootElem.appendChild(selectMenu);
+}
+function createSearchBar() {
+  const rootElem = document.getElementById("root");
+
+  //search bar
+  const searchBar = document.createElement("input");
+  searchBar.id = "txtSearchBy";
+  searchBar.setAttribute("type", "string");
+  searchBar.addEventListener("input", function () {
+    drawEpiCard();
+  });
+
+  //search result
+  const searchResult = document.createElement("label");
+  searchResult.id = "searchResult";
+
+  // const resultP = document.createElement("p");
+  // resultP.innerHTML = `Got ${_localEpisodes.length} episode(s). The data has (originally) come from <a href= "https://www.tvmaze.com/">TVMaze.com</a><br>`;
+  // //rootElem.innerHTML = `Got ${_localEpisodes.length} episode(s). The data has (originally) come from <a href= "https://www.tvmaze.com/">TVMaze.com</a><br>`;
+
+  // rootElem.insertBefore(resultP, rootElem.firstChild);
+  rootElem.appendChild(searchBar);
+  rootElem.appendChild(searchResult);
+  //drawEpiCard();
+}
+function clearCards() {
+  document
+    .querySelectorAll("#root div.grid-container")
+    .forEach((el) => el.remove());
+}
+function drawEpiCard() {
```

Comment:
as this function relies on `_localEpisodes` it would be good to pass this in as argument, so like `function drawEpisodeCards(allEpisodes) {`